protostar/migrator/maybe_tmp_directory.py

```diff
@@ -0,0 +1,36 @@
+from contextlib import AbstractContextManager
+from os import listdir, rmdir
+from pathlib import Path
+from typing import Union
+
+from protostar.protostar_exception import ProtostarException
+
+
+class MaybeTmpDirectory(AbstractContextManager):
+    def __init__(self, path: Path):
+        self._path = path
+
+    def __enter__(self) -> None:
+        self._create_dir()
```

Comment:
I would return `self._path` here so that usage places could do:

```python
with MaybeTmpDirectory(tmp_path / "tmp_dir") as output_path:
    ...
```